src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -118,4 +122,42 @@ void AsyncOpenTask::unregister_download_progress_notifier(uint64_t token)
         m_session->unregister_progress_notifier(token);
 }
 
+void AsyncOpenTask::run_subscription_initializer(SubscriptionCallback&& subscription_callback,
+                                                 AsyncOpenCallback&& async_open_callback,
+                                                 std::shared_ptr<_impl::RealmCoordinator> coordinator,
+                                                 bool rerun_subscription_on_open)
+{
+    auto shared_realm = coordinator->get_realm();
+    auto latest_subscription = shared_realm->get_latest_subscription_set();
+    if (latest_subscription.version() == 0 || rerun_subscription_on_open) {
```

Comment:
we'll have to change the logic here as per [Desi's comment](https://github.com/realm/realm-core/issues/5962#issuecomment-1603728769)